### PR TITLE
Set up pylint workflow

### PR DESCRIPTION
**Pull request by baimamboukar**
- A `pylint` workflow to run python code analysis
- Added workflow badge in README
- Applied linters suggestions (achieve 10.0/10 code rating 🎉)

### CODE REVIEW

File: .github/workflows/pylint.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Pylint Code analysis
+
+on:
+  push:
+    branches: ["*"]
```
Comment (marcin-krystianc) [inline .github/workflows/pylint.yaml:5]: nit: I think specifying branch here is not necessary (the default is to run on all branches)

File: .github/workflows/pylint.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+    branches: ["*"]
```
Comment (marcin-krystianc) [inline .github/workflows/pylint.yaml:7]: same here

**Review by marcin-krystianc — APPROVED** [review]
Looks, good 👍 
Just a minor comment about unnecessary branch specification. 
Feel free to ignore though.

**Comment by baimamboukar** [comment]
> Looks, good 👍 Just a minor comment about unnecessary branch specification. Feel free to ignore though.

Hello @marcin-krystianc ! Thanks for your feedback

File: .github/workflows/pylint.yaml (hunk 1 above)
Comment (baimamboukar) [reply .github/workflows/pylint.yaml:5]: Yes! I think this is what I did by setting the `branches:` to `*`

It should run on all branches....

Or you want me to update something ? @baimamboukar

Comment (marcin-krystianc) [reply .github/workflows/pylint.yaml:5]: I think you can remove `branches: ["*"]` entry altogether - it will work exactly the same

Comment (baimamboukar) [reply .github/workflows/pylint.yaml:5]: alright!... I'm on it

Comment (baimamboukar) [reply .github/workflows/pylint.yaml]: It is done ✅🚀

**Pull request merged**